components/tikv_util/src/memory.rs

```diff
@@ -211,23 +223,22 @@ impl MemoryQuota {
         self.capacity.load(Ordering::Relaxed)
     }
 
-    pub fn set_capacity(&self, mut capacity: usize) {
-        // Value bigger than isize::MAX just means unlimited,
-        // so replace it with isize::MAX to avoid overflow.
-        if capacity > isize::MAX as usize {
-            capacity = isize::MAX as usize;
-        }
+    pub fn set_capacity(&self, capacity: usize) {
+        let capacity = Self::adjust_capacity(capacity);
         self.capacity.store(capacity, Ordering::Relaxed);
     }
 
     pub fn alloc_force(&self, bytes: usize) {
+        if bytes > MAX_MEMORY_ALLOC_SIZE {
+            return;
```

Comment:
Can we add some log or metrics here?

Reply:
Done. Add a warn log for oversized alloc.  Maybe we can also add a metrics for the alloc size distribution histogram later if we really need it.